jupyter_remote_desktop_proxy/install.py

```diff
@@ -0,0 +1,151 @@
+                md5 = media.find('{http://video.search.yahoo.com/mrss/}hash').text
+                break
+        else:
+            raise Exception(f"Could not find a tarball \"{tigervnc_archive}\" on sourceforge.net")
+    return url, md5
+
+
+def install_tigervnc(prefix, plat, tigervnc_version):
+    tigervnc_path = os.path.join(prefix, "tigervnc")
+
+    tigervnc_archive = f"tigervnc-{tigervnc_version}.{plat}.tar.gz"
+    tigervnc_archive_path = os.path.join(prefix, tigervnc_archive)
+
+    if os.path.exists(tigervnc_path):
+        print(f"Tigervnc already exists at {tigervnc_path}. Remove it to re-install.")
+        print("--- Done ---")
+        return
+
+    try:
+        tigervnc_url, tigervnc_md5 = fetch_release_info(tigervnc_archive, tigervnc_version)
+    except HTTPError as e:
+        print(f"Failed to retrieve md5 and download url: {e}")
+        return
+    except Exception as e:
+        print(f"Failed to retrieve md5 and download url: {e}")
+        return
+
+    print(f"Downloading tigervnc {tigervnc_version} from {tigervnc_url}...")
+    urlretrieve(tigervnc_url, tigervnc_archive_path)
+
+    md5 = checksum_file(tigervnc_archive_path)
+    if md5 != tigervnc_md5:
+        raise OSError(f"Checksum failed {md5} != {tigervnc_md5}")
+
+    print("Extracting the archive...")
+    with tarfile.open(tigervnc_archive_path, "r") as tar_ref:
+        tar_ref.extractall(prefix)
+
+    shutil.move(f"{prefix}/tigervnc-{tigervnc_version}.{plat}/usr/bin", prefix)
+    shutil.move(f"{prefix}/tigervnc-{tigervnc_version}.{plat}/usr/lib64", prefix)
+    shutil.move(f"{prefix}/tigervnc-{tigervnc_version}.{plat}/usr/share", prefix)
+
+    print(f"Installed tigervnc {tigervnc_version}")
+    shutil.rmtree(f"{prefix}/tigervnc-{tigervnc_version}.{plat}")
+    os.unlink(tigervnc_archive_path)
+    print("--- Done ---")
+
+
+def main():
+    # extract supported and default versions from urls
+    parser = argparse.ArgumentParser(
+        description="Dependency installer helper",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    parser.add_argument(
+        "--output",
+        dest="installation_dir",
+        default=f"{HERE}/share/tigervnc",
+        help=textwrap.dedent(
+            """\
+            The installation directory (absolute or relative path).
+            If it doesn't exist, it will be created.
+            If no directory is provided, it defaults to:
+            --- %(default)s ---
+            """
+        ),
+    )
+
+    machine = platform.machine()
+    parser.add_argument(
+        "--platform",
+        dest="plat",
+        default=machine,
+        help=textwrap.dedent(
+            """\
+            The platform to download for.
+            If no platform is provided, it defaults to:
+            --- %(default)s ---
+            """
+        ),
+    )
+
+    parser.add_argument(
+        "--tigervnc-version",
+        dest="tigervnc_version",
+        default="1.10.0",
```

Comment:
Does the current version `1.13.0` work?

Reply:
So far, I have been unable to start a desktop with tigervnc > 1.10.0. `vncserver` was not only moved from `bin` to `libexec` but also deeply modified to the point where in tigervnc 1.13.0 it only accepts a display number.

But there might be hope. The `vncserver` perl script only purpose is to craft an `Xvnc` command-line with arguments and call it. Therefore, I think it might be possible to skip calling `vncserver` and call `Xvnc` instead since we already in the "command-line crafting business" with jupyter-remote-desktop-proxy. This assumes that `Xvnc` arguments were pretty much the same between tigervnc 1.10.0 and 1.13.0, which might be a strong hypothesis.

Reply:
That's a pain, thanks for investigating. If it's easier then leave it as is, and we can come back to it in a future PR.

Reply:
I have added a list of valid choices for `--tigervnc-version` to avoid issue with users selecting anything above 1.10.1.